### ISSUE TITLE
inertia-adonisjs 8.0.0 is not working, 7.4.0 is working

### ISSUE THREAD

**Issue by jitterbux** [eidellev/inertiajs-adonisjs#113]
It kind of feels like everything is broken, the code and the docs for the version 8. Should we go back to 7.4.0?
Or do you think you will fix it in a week or two?
Especially the SSR seems not to be working at all.
Perhaps a working Vue example project would help?

**Comment by jitterbux** [comment]
Yeah, I am stuck at this error message: `Instead rename index.js to end in .cjs, change the requiring code to use dynamic import() which is available in all CommonJS modules` so I am not sure how to fix it. It points to this line: `module.exports = require("@inertiajs/vue3");
` in ssr.js.

**Comment by jitterbux** [comment]
Were you able to make your Adonis project work with Inertia 1.0 and especially SSR? Could you please share such project on Github. Perhaps I and others could study and compare the code file by file and it will be even better for you not to explain all the things over and over. Because I can't find a working example of Adonis and Inertia 1.0 with SSR. It would be cool if you could create some repo with the basic working example of Adonis and Inertia with SSR, preferably using Vue ;) Thank you a lot in advance.

**Comment by mastermunj** [comment]
@jitterbux Please try updating `webpack.ssr.config.js` as follows:


```js
const nodeExternals = require('webpack-node-externals');
```

Then replace 
```js
config.externals = [require('webpack-node-externals')()]
```

with

```js
config.externals = [
  nodeExternals({
    allowlist: ['@inertiajs/core', '@inertiajs/vue3'],
  }),
];
```

This solved the problem for me.

**Comment by Quent1Pr** [comment]
Thank you @mastermunj I had the same issue but after that I have this error "Cannot use import statement outside a module" (with react)

@eidellev  I agree with @jitterbux It's hard to find a working example with ssr ! I saw @RomainLanz trying it on Youtube.

**Comment by RomainLanz** [comment]
I have made it work with Vue and React. I will check if I can improve the readme, but I don't recall having any issues with this package.

The only thing to consider is that you should not use Inertia 1.0 yet.

**Comment by geosemaan** [comment]
8.0.1 seems to work just fine with Inertia 1.0

I came across this tutorial on how to set it up and it works just fine. I didn't even enable SSR (skip that part).

Also make sure to install and configure `view` and `session` before continuing.

https://blog.logrocket.com/getting-started-inertia-js-vue-js-adonisjs/

**Comment by ammezie** [comment]
@RomainLanz curious as to why Inertia 1.0 shouldn't be used yet? I thought it's a stable release.

**Comment by RomainLanz** [comment]
Because this package has not been updated for it yet. No issue with Inertia 1.0 directly 😄

**Comment by ammezie** [comment]
Oh! Makes sense.

**Comment by brlebtag** [comment]
> I have made it work with Vue and React. I will check if I can improve the readme, but I don't recall having any issues with this package.
> 
> The only thing to consider is that you should not use Inertia 1.0 yet.

Could you share a "how to"? I am trying to use Adonis + SSR + React but I have not succeeded so far.

**Comment by RomainLanz** [comment]
> Could you share a "how to"? I am trying to use Adonis + SSR + React but I have not succeeded so far.

If you want to enable SSR, you have to use version `7.4.0` of this package, and you must code your frontend in CJS.

Using the latest version will move your frontend code to ESM. Doing so will require to also have an ESM backend to render your components/pages, and this is not the case with AdonisJS 5.

TL;DR: If you want to do SSR with AdonisJS 5 and Inertia, use version `7.4.0` and code in CJS.
